src/concepts/pipeline/filtering.md

```diff
@@ -0,0 +1,105 @@
+
+What - What happens to the data when filtered here
+
+When - Which situations this filtering should be, and is, used in
+
+How - What kind of data can be filtered at this stage
+
+### Edge filtering
+
+Where: Filtered by nginx, [currently we use it to filter out non-v4 pings](https://github.com/mozilla-services/cloudops-infra/blob/master/projects/data-ingestion/k8s/charts/data-ingestion/templates/filter-configmap.yaml#L14-L41)
+([to be removed, but capability to remain](https://bugzilla.mozilla.org/show_bug.cgi?id=1678497)).
+
+What: Drops data entirely from the pipeline; there will be no traces of it downstream from the edge server.
+
+When: Only to be used in extreme situations (e.g. PII exposure). We also use it for dropping [too-large messages](https://github.com/mozilla/gcp-ingestion/blob/master/docs/architecture/overview.md#limits) and [headers](https://github.com/mozilla/gcp-ingestion/blob/master/ingestion-edge/ingestion_edge/util.py#L95).
+
+How: Can be used to filter by URI namespaces, apps, etc.; but not anything in the payload. 
+
+### Beam Filtering
+
+Where: Filtered in the [message scrubber](https://github.com/mozilla/gcp-ingestion/blob/master/ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/MessageScrubber.java).
+
+What: Causes data to be written to the error stream or to be dropped entirely.
+
+When: Filter out data we absolutely know we will never need to see (e.g. data from forked applications).
+
+How: Can filter out namespaces, doctypes, or URIs currently; in the extreme can filter on any [message Attribute](https://github.com/mozilla/gcp-ingestion/blob/master/ingestion-core/src/main/java/com/mozilla/telemetry/ingestion/core/Constant.java#L8) or payload field.
+
+### JSON Schema Filtering
+
+Where: During ingestion, as defined in the [payload schema](https://github.com/mozilla-services/mozilla-pipeline-schemas/).
+
+What: Causes data to be written to the error stream.
+
+When: When trying to remove bad analysis data that we know we will never need (e.g. huge values, improper strings, etc.). Usually these indicate something went wrong with the payload.
+
+How: Can filter on values in the payload, using the [JSON schema](https://json-schema.org/understanding-json-schema/).
+
+### Filtering from the stable tables
+
+Where: After ingestion to live tables, but [before copying to the stable tables](https://github.com/mozilla/bigquery-etl/blob/master/bigquery_etl/copy_deduplicate.py#L40).
+
+What: Allows data to exist in the live tables, but removes it from the stable tables.
+
+When: Use for data that may be needed for some analyses on recent data, but not for data that will need long-term historical analyses or for use in any downstream reporting. For example, we filter out pings from automation (e.g. CI) here, so that analysis is unaffected by them, but we can still analyze what the recent CI data looks like. We also drop duplicate pings (per the document-id).
```

Comment:
May wish to link to [the docs in the Book about automation filtering](https://mozilla.github.io/glean/book/user/debugging/index.html?highlight=sourcetags#enabling-debugging-features-through-environment-variables) here.